src/list_function.py

```diff
@@ -13,24 +13,33 @@
 
 import os
 from http import HTTPStatus
+from datetime import timedelta
 
 import boto3
 
 import aws_lambda_api_event_utils as api_utils
-from aws_error_utils import errors, ClientError
+from aws_error_utils import errors, catch_aws_error, ClientError
 
 from common.identifiers import parse_key
 from common.pagination import (
     get_encryption_client,
     encode_pagination_token,
     decode_pagination_token,
+    pagination_token_context_from_event,
 )
 
 print("initializing")
 
 TABLE_NAME = os.environ["TABLE_NAME"]
 PAGINATION_KEY_ARN = os.environ["PAGINATION_KEY_ARN"]
 
+PAGINATION_TOKEN_VALIDITY_DURATION = timedelta(minutes=5)
+if "PAGINATION_TOKEN_VALIDITY_DURATION_SECONDS" in os.environ:
+    # fail loudly if we can't parse the duration
+    PAGINATION_TOKEN_VALIDITY_DURATION = timedelta(
+        seconds=float(os.environ["PAGINATION_TOKEN_VALIDITY_DURATION_SECONDS"])
+    )
```

Comment:
Off-topic: one of the things I like about Go is that it has a [built-in duration parser](https://pkg.go.dev/time#ParseDuration) so that a person doesn't need to calculate how many seconds they need if they want a 15-minute duration, for example (I wish I didn't have various increments burned into my brain). I saw a [neat implementation of something similar recently for Python](https://gist.github.com/benkehoe/5b03c308b038b29e42106f602e554010), you might be familiar with it. Thoughts on applicability here?

Reply:
Sure, if you wanted to pull that code in, we could use it here.